### PR TITLE
add unstructured table element node parser 

### PR DESCRIPTION
**Pull request by jerryjliu**
this notebook is able to parse any HTML document object into a set of nodes. 

usage is similar to our node parsers here: #7863

includes a `get_base_nodes_and_mappings` function so that users can extract our the "base" nodes + underlying table nodes, and setup recursive retrieval

### CODE REVIEW

File: llama_index/node_parser/unstructured_element.py [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+
+    Splits a document into Text Nodes and Index Nodes corresponding to embedded objects
+    (e.g. tables).
+
+    """
+
+    callback_manager: CallbackManager = Field(
+        default_factory=CallbackManager, exclude=True
+    )
+    llm: Optional[Any] = Field(
```
Comment (logan-markewich) [inline llama_index/node_parser/unstructured_element.py:244]: Should this be `Optional[LLM]` ?

File: llama_index/node_parser/unstructured_element.py [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+    # flush text buffer
+    if len(cur_text_el_buffer) > 0:
+        cur_text_nodes = _get_nodes_from_buffer(cur_text_el_buffer, node_parser)
+        nodes.extend(cur_text_nodes)
+        cur_text_el_buffer = []
+
+    return nodes
+
+
+def get_base_nodes_and_mappings(nodes: List[BaseNode]) -> Tuple[List[BaseNode], Dict]:
```
Comment (logan-markewich) [inline llama_index/node_parser/unstructured_element.py:198]: Any reason to make these functions outside of the class? Are they still used in the notebook?

Comment (jerryjliu) [reply llama_index/node_parser/unstructured_element.py:198]: oh hm no particular reason. Maybe I'll move inside the class so users don't need to import two things

File: llama_index/node_parser/unstructured_element.py (hunk 1 above)
Comment (jerryjliu) [reply llama_index/node_parser/unstructured_element.py:244]: i was going to make an argument about circular deps but realized we already top-level import OpenAI, so i'll just type this as LLM

**Pull request merged**